shtdlib.sh

```diff
@@ -157,6 +157,26 @@ function debug {
     fi
 }
 
+# Umask decorator, changes the umask for a function
+# To use this add a line like the following (without #) as the first line of a function
+# umask_decorator "${FUNCNAME[0]}" "${@}" && return
+# To specify a different umask set the umask_decorator_mask variable to the
+# desired umask.
+function umask_decorator {
+    if [ "${FUNCNAME[0]}" != "${FUNCNAME[2]}" ] ; then
+        mask="${umask_decorator_mask:-0007}"
```

Comment:
Should make this a local variable in case we use a `mask` variable elsewhere.

Reply:
Somehow I thought all variables defined inside functions were local, changed.